spark/src/main/scala/org/apache/spark/sql/delta/commands/OptimizeTableCommand.scala

```diff
@@ -239,7 +239,9 @@ class OptimizeExecutor(
 
   private val isClusteredTable = ClusteredTableUtils.isSupported(txn.snapshot.protocol)
 
-  private val isMultiDimClustering = isClusteredTable || zOrderByColumns.nonEmpty
+  private val isMultiDimClustering =
+    optimizeStrategy.isInstanceOf[ClusteringStrategy] ||
+    optimizeStrategy.isInstanceOf[ZOrderStrategy]
```

Comment:
Does this fix something or just a good refactoring?

Reply:
We need both this fix and the fix in OptimizeTableStrategy:
* isMultiDimClustering: controls whether we run `MultiDimClustering.cluster` or compaction ([pointer](https://github.com/delta-io/delta/blob/a11b92de2850dfbe439fad6ec7cca747be503b77/spark/src/main/scala/org/apache/spark/sql/delta/commands/OptimizeTableCommand.scala#L423))
* Fix in getMode in OptimizeTableStrategy: to determine the correct mode

Reply:
Note that the previous logic here was wrong:
`isClusteredTable || zOrderByColumns.nonEmpty` should be
`(isClusteredTable && hasClusteringColumns) || zOrderByColumns.nonEmpty`